streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableImplTest.java

```diff
@@ -350,27 +338,27 @@ public void testRepartition() throws NoSuchFieldException, IllegalAccessExceptio
         table1.groupBy(MockMapper.<String, String>noOpKeyValueMapper())
             .reduce(MockReducer.STRING_ADDER, MockReducer.STRING_REMOVER, Materialized.<String, String, KeyValueStore<Bytes, byte[]>>as("mock-result2"));
 
-        driver.setUp(builder, stateDir, stringSerde, stringSerde);
-        driver.setTime(0L);
+        try (final TopologyTestDriverWrapper driver = new TopologyTestDriverWrapper(builder.build(), props)) {
 
-        // three state store should be created, one for source, one for aggregate and one for reduce
-        assertEquals(3, driver.allStateStores().size());
+            // three state store should be created, one for source, one for aggregate and one for reduce
+            assertEquals(3, driver.getAllStateStores().size());
 
-        // contains the corresponding repartition source / sink nodes
-        assertTrue(driver.allProcessorNames().contains("KSTREAM-SINK-0000000003"));
-        assertTrue(driver.allProcessorNames().contains("KSTREAM-SOURCE-0000000004"));
-        assertTrue(driver.allProcessorNames().contains("KSTREAM-SINK-0000000007"));
-        assertTrue(driver.allProcessorNames().contains("KSTREAM-SOURCE-0000000008"));
+            // contains the corresponding repartition source / sink nodes
+            assertTrue(driver.getAllProcessorNames().contains("KSTREAM-SINK-0000000003"));
```

Comment:
I am wondering if we should rewrite this using `Topology#describe()` instead of hooking into the test driver?

Actually same thought about `driver.getAllStateStores()` ?

Reply:
That's a good idea. `allProcessorNames` is only used in this test, and its functionally is easily replicated by using `Topology#describe()`.
 
In the case of `allStateStores`, it provides a bit more functionality than `allProcessorNames` since it returns a `Map<String, StateStore>`. It is also part of the public API. Getting all processor names could be done using `TopologyWrapper` to access the `InternalTopologyBuilder`, so it could in principle be removed. However, it could be useful for end users' testing, which do not have access to `TopologyWrapper`, so it might be nice to keep it around.